### PR TITLE
Fix incompatibilities with upcoming security 6.0

### PR DESCRIPTION
**Pull request by wouterj**
| Q             | A
| ------------- | ---
| Branch?       | 5.4
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | yes
| Tickets       | -
| License       | MIT
| Doc PR        | -

Hats of to the person that invented the flipped tests on a new major branch! All these incompatibility bugs were discovered by the flipped tests of #41613

### CODE REVIEW

File: UPGRADE-5.4.md [hunk 1]
```diff
@@ -66,6 +66,7 @@ SecurityBundle
 Security
 --------
 
+ * Deprecate `AuthenticationEvents::AUTHENTICATION_FAILURE`, use the `LoginFailureEvent` instead
```
Comment (wouterj) [inline UPGRADE-5.4.md:69]: This event has been deprecated in 5.3, but we forgot to deprecate this constant.

**Comment by wouterj** [comment]
Build should be green after #42596 is merged (and this PR is rebased).

**Comment by fabpot** [comment]
@wouterj  This one can now be rebased.

**Comment by wouterj** [comment]
Check, done (and ~hopefully~ also fixed the 5.4 branch in this PR)

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @wouterj.

**Pull request merged**